subsys/usb/usb_device.c

```diff
@@ -792,6 +803,35 @@ static bool usb_handle_std_device_req(struct usb_setup_packet *setup,
 	return ret;
 }
 
+/**
+ * @brief Check if the interface of given number is valid
+ *
+ * @param [in] interface Number of the addressed interface
+ *
+ * This function searches through descriptor and checks
+ * is the Host has addressed valid interface.
+ *
+ * @return true if interface exists - valid
+ */
+static bool is_interface_valid(uint8_t interface)
+{
+	const uint8_t *p = (uint8_t *)usb_dev.descriptors;
+	const struct usb_if_descriptor *if_descr;
+
+	/* Search through descriptor for matching interface */
+	while (p[DESC_bLength] != 0U) {
```

Comment:
That can be simplified, just check `setup->wIndex`(L853) against bNumInterfaces from [Configuration Descriptor](https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/usb/usb_descriptor.c#L92)

Reply:
That's good point. I will test it.

Reply:
Done.